pymatgen/command_line/bader_caller.py

```diff
@@ -15,19 +15,19 @@
 from __future__ import annotations
 
 import os
-import shutil
 import subprocess
 import warnings
 from glob import glob
 from shutil import which
 from tempfile import TemporaryDirectory
 
 import numpy as np
-from monty.io import zopen
+from monty.tempfile import ScratchDir
```

Comment:
I actually don't understand why the files need to be copied at all? Does `bader` modify them in place? Seems unlikely and even if, they should provide an option to disable that?

Reply:
They don't! So what I did here is creating a temporary scratchdir to store the output files from bader. However, bader cannot read compressed file so at the bottom we need to decompress file if it is compressed and store it temporarily somewhere (I just put it in ScratchDir as well)